utils/calendrical_calculations/src/persian.rs

```diff
@@ -55,7 +77,24 @@ pub fn arithmetic_persian_from_fixed(date: RataDie) -> Result<(i32, u8, u8), I32
     Ok((year, month, day))
 }
 
+/// arithmetic_persian_from_fixed, modified to use the 33-year rule method
+pub fn fast_persian_from_fixed(date: RataDie) -> Result<(i32, u8, u8), I32CastError> {
+    let year = fast_persian_year_from_fixed(date);
+    let year = i64_to_i32(year)?;
+    #[allow(clippy::unwrap_used)] // valid month,day
```

Comment:
Observation: I think you don't need this suppression. There is no unwrap in the next line

Reply:
Remvoed.